feature/s3/transfermanager/api_op_GetObject.go

```diff
@@ -0,0 +1,940 @@
+}
+
+func (o *GetObjectOutput) mapFromGetObjectOutput(out *s3.GetObjectOutput, checksumMode s3types.ChecksumMode) {
+	o.AcceptRanges = aws.ToString(out.AcceptRanges)
+	o.CacheControl = aws.ToString(out.CacheControl)
+	o.ChecksumMode = types.ChecksumMode(checksumMode)
+	o.ChecksumCRC32 = aws.ToString(out.ChecksumCRC32)
+	o.ChecksumCRC32C = aws.ToString(out.ChecksumCRC32C)
+	o.ChecksumSHA1 = aws.ToString(out.ChecksumSHA1)
+	o.ChecksumSHA256 = aws.ToString(out.ChecksumSHA256)
+	o.ContentDisposition = aws.ToString(out.ContentDisposition)
+	o.ContentEncoding = aws.ToString(out.ContentEncoding)
+	o.ContentLanguage = aws.ToString(out.ContentLanguage)
+	o.ContentRange = aws.ToString(out.ContentRange)
+	o.ContentType = aws.ToString(out.ContentType)
+	o.ETag = aws.ToString(out.ETag)
+	o.Expiration = aws.ToString(out.Expiration)
+	o.ExpiresString = aws.ToString(out.ExpiresString)
+	o.Restore = aws.ToString(out.Restore)
+	o.SSECustomerAlgorithm = aws.ToString(out.SSECustomerAlgorithm)
+	o.SSECustomerKeyMD5 = aws.ToString(out.SSECustomerKeyMD5)
+	o.SSEKMSKeyID = aws.ToString(out.SSEKMSKeyId)
+	o.VersionID = aws.ToString(out.VersionId)
+	o.WebsiteRedirectLocation = aws.ToString(out.WebsiteRedirectLocation)
+	o.BucketKeyEnabled = aws.ToBool(out.BucketKeyEnabled)
+	o.DeleteMarker = aws.ToBool(out.DeleteMarker)
+	o.MissingMeta = aws.ToInt32(out.MissingMeta)
+	o.PartsCount = aws.ToInt32(out.PartsCount)
+	o.TagCount = aws.ToInt32(out.TagCount)
+	o.ContentLength = aws.ToInt64(out.ContentLength)
+	o.Body = out.Body
+	o.Expires = aws.ToTime(out.Expires)
+	o.LastModified = aws.ToTime(out.LastModified)
+	o.ObjectLockRetainUntilDate = aws.ToTime(out.ObjectLockRetainUntilDate)
+	o.Metadata = out.Metadata
+	o.ObjectLockLegalHoldStatus = types.ObjectLockLegalHoldStatus(out.ObjectLockLegalHoldStatus)
+	o.ObjectLockMode = types.ObjectLockMode(out.ObjectLockMode)
+	o.ReplicationStatus = types.ReplicationStatus(out.ReplicationStatus)
+	o.RequestCharged = types.RequestCharged(out.RequestCharged)
+	o.ServerSideEncryption = types.ServerSideEncryption(out.ServerSideEncryption)
+	o.StorageClass = types.StorageClass(out.StorageClass)
+	o.ResultMetadata = out.ResultMetadata.Clone()
+}
+
+// GetObject downloads an object from S3, intelligently splitting large
+// files into smaller parts/ranges according to config and getting them in sequence.
+// You can configure the download type and chunk size through the Options parameters.
+//
+// Additional functional options can be provided to configure the individual
+// download. These options are copies of the original Options instance, the client of which GetObject is called from.
+// Modifying the options will not impact the original Client and Options instance.
+func (c *Client) GetObject(ctx context.Context, input *GetObjectInput, opts ...func(*Options)) (*GetObjectOutput, error) {
+	i := downloader{in: input, options: c.options.Copy()}
+	for _, opt := range opts {
+		opt(&i.options)
+	}
+	i.options.Concurrency = 1
```

Comment:
Why are we setting Concurrency to 1?

Reply:
`GetObject` here mimics `s3.GetObject` to get an object's chunks and write to output body in sequence, thus there should be only one goroutine to write chunks to output in order

Reply:
You'll need to benchmark it but I don't think we can do it this way. Doing multiple range GETs w/ concurrency 1 is going to be actively worse than just plain GetObject, in theory, because you're adding the overhead of multiple round-trips.

Ideally we dispatch concurrent range GETs and then we just arrange their output streams in sequence behind a single `io.ReadCloser` to the user.